src/commands/commands.cpp

```diff
@@ -0,0 +1,49 @@
+#include "commands.h"
+
+Commands::Commands() {
+    cmdList = {};
+}
+
+// TODO: destory command list on close
+Commands::~Commands() {
+}
+
+void Commands::add(const CommandInfo* command) {
+    cmdList[command->name] = command;
+}
+
+void Commands::add(const std::string name, const std::string usage, const commandFunction* func) {
+    const CommandInfo info = {
+            .name = name,
+            .usage = usage,
+            .command = func
+    };
+
+    cmdList[name] = &info;
+}
+
+
+const CommandInfo* Commands::get(const std::string name) {
+    if(cmdList.find(name) == cmdList.end()) {
+        return nullptr;
```

Comment:
we can return `std::optional` instead of using pointers for 'maybe returns value'